src/python/pants/init/logging.py

```diff
@@ -73,79 +46,120 @@ def formatException(self, exc_info):
         return "\n(Use --print-stacktrace to see more error details.)"
 
 
-def clear_logging_handlers():
-    logger = logging.getLogger(None)
-    for handler in get_logging_handlers():
-        logger.removeHandler(handler)
+@contextmanager
+def stdio_destination(stdin_fileno: int, stdout_fileno: int, stderr_fileno: int):
```

Comment:
This returns `-> Iterator[None]`, which is helpful to mark so people know not to use `as _` with the context manager.